lib/src/native/realm_core.dart

```diff
@@ -1502,6 +1502,28 @@ class _RealmCore {
     _realmLib.realm_sync_session_unregister_progress_notifier(session.handle._pointer, token);
   }
 
+  int sessionRegisterConnectionStateNotifier(Session session, SessionConnectionStateController controller) {
+    return _realmLib.realm_dart_sync_session_register_connection_state_change_callback(
+        session.handle._pointer,
+        Pointer.fromFunction(on_connection_state_change),
+        controller.toPersistentHandle(),
```

Comment:
My usual reservations

Reply:
Yes this should be a weakhandle. Since we not doing that for the other callback lets have a //TODO here to fix it